### PR TITLE
feat(incarnation-scafi): remove ScafiNode to support the new NodeProperty API

### PR DESCRIPTION
**Pull request by cric96**

### CODE REVIEW

File: alchemist-incarnation-scafi/src/main/scala/it/unibo/alchemist/model/implementations/actions/SendScafiMessage.scala [hunk 1]
```diff
@@ -11,14 +11,14 @@ package it.unibo.alchemist.model.implementations.actions
...
-import it.unibo.alchemist.model.implementations.nodes.ScafiNode
 import it.unibo.alchemist.model.interfaces._
+
 import java.util.stream.Collectors
 import scala.jdk.CollectionConverters._
 
 class SendScafiMessage[T, P<:Position[P]](
   env: Environment[T, P],
-  node: ScafiNode[T, P],
+  node: Node[T],
```
Comment (DanySK) [inline alchemist-incarnation-scafi/src/main/scala/it/unibo/alchemist/model/implementations/actions/SendScafiMessage.scala:21]: Why not `ScafiDevice` directly?

File: alchemist-incarnation-scafi/src/main/scala/it/unibo/alchemist/model/implementations/actions/SendScafiMessage.scala [hunk 2]
```diff
@@ -30,17 +30,16 @@ class SendScafiMessage[T, P<:Position[P]](
...
-   * @param n
+   * @param destinationNode
    * The node where to clone this { @link Action}
    * @param r
    * The reaction to which the CURRENT action is assigned
    * @return the cloned action
    */
-  override def cloneAction(n: Node[T], r: Reaction[T]): Action[T] = {
-    if(!isScafiNode(n)) {
+  override def cloneAction(destinationNode: Node[T], r: Reaction[T]): Action[T] = {
```
Comment (DanySK) [inline alchemist-incarnation-scafi/src/main/scala/it/unibo/alchemist/model/implementations/actions/SendScafiMessage.scala:39]: r -> reaction

File: alchemist-incarnation-scafi/src/main/scala/it/unibo/alchemist/model/ScafiIncarnation.scala [hunk 3]
```diff
@@ -84,12 +82,11 @@ sealed class ScafiIncarnation[T, P <: Position[P]] extends Incarnation[T, P]{
...
   }
 
-  override def createConcentration(): T = createConcentration("")
+  override def createConcentration(): T = null.asInstanceOf[T]
 
   override def createCondition(rand: RandomGenerator, env: Environment[T, P] , node: Node[T], time: TimeDistribution[T], reaction: Reaction[T], param: String): Condition[T] = {
     if(!isScafiNode(node)) {
-      throw new IllegalArgumentException(s"The node must be an instance of ${classOf[ScafiNode[_,_]]}"
-      + s", but it is an ${node.getClass} instead.")
+      throw new IllegalArgumentException(s"The node must has an instance of ${classOf[ScafiDevice[_]]} as property")
```
Comment (DanySK) [inline alchemist-incarnation-scafi/src/main/scala/it/unibo/alchemist/model/ScafiIncarnation.scala:89]:
```suggestion
      throw new IllegalArgumentException(s"The node must have a ${classOf[ScafiDevice[_]].getSimpleName} property")
```

File: alchemist-incarnation-scafi/src/main/scala/it/unibo/alchemist/model/ScafiIncarnation.scala [hunk 4]
```diff
@@ -84,12 +82,11 @@ sealed class ScafiIncarnation[T, P <: Position[P]] extends Incarnation[T, P]{
     CachedInterpreter[AnyRef](if(doCacheValue) v else v.tail, doCacheValue).asInstanceOf[T]
   }
 
-  override def createConcentration(): T = createConcentration("")
+  override def createConcentration(): T = null.asInstanceOf[T]
```
Comment (DanySK) [inline alchemist-incarnation-scafi/src/main/scala/it/unibo/alchemist/model/ScafiIncarnation.scala:85]: ok to me, but that sounds a lot NPE-prone.

**Review by DanySK — CHANGES_REQUESTED** [review]
Minor changes

File: alchemist-incarnation-scafi/src/main/scala/it/unibo/alchemist/model/implementations/actions/SendScafiMessage.scala (hunk 2 above)
Comment (cric96) [reply alchemist-incarnation-scafi/src/main/scala/it/unibo/alchemist/model/implementations/actions/SendScafiMessage.scala]: Ops... I fixed this! Furthermore, while I was at it, I tried to follow clean code guidelines for naming convention in the entire scafi incarnation.. :)

**Comment by DanySK** [comment]
@cric96 this PR will be auto-squashed-in as soon as the criteria are met.

File: alchemist-incarnation-scafi/src/main/scala/it/unibo/alchemist/model/ScafiIncarnation.scala (hunk 4 above)
Comment (cric96) [reply alchemist-incarnation-scafi/src/main/scala/it/unibo/alchemist/model/ScafiIncarnation.scala:85]: Without that, there are issues about molecule initialisation.. Indeed, in `ProtelisIncarnation` `createConcentration()` returns null too ([link](https://github.com/AlchemistSimulator/Alchemist/blob/6baf5dcf38fcaa0d4ba4c3be7c6e77969fa9490c/alchemist-incarnation-protelis/src/main/java/it/unibo/alchemist/model/ProtelisIncarnation.java#L185))

File: alchemist-incarnation-scafi/src/main/scala/it/unibo/alchemist/model/implementations/actions/SendScafiMessage.scala (hunk 1 above)
Comment (cric96) [reply alchemist-incarnation-scafi/src/main/scala/it/unibo/alchemist/model/implementations/actions/SendScafiMessage.scala]: Partially addressed in this [commit](https://github.com/AlchemistSimulator/Alchemist/pull/1300/commits/7bab1d016458adb474632259c3d2e8ecfac54943).

**Pull request merged**